pytorch_lightning/core/saving.py

```diff
@@ -23,20 +24,20 @@ def on_save_checkpoint(self, checkpoint):
     # -------------------------
     # OPTIONAL HOOKS
     # -------------------------
-    def on_hpc_save(self, checkpoint):
+    def on_hpc_save(self, checkpoint: dict):
```

Comment:
those can be also `-> None` right?

Reply:
I thought about it, but was not sure if it is best practice to annotate `-> None` for void functions/methods. What do you think, should we always do it?

Reply:
@hadim what is your thought?

Reply:
For return types, I think `None` is as informative as any other type so yes I would add it.

Reply:
thanks for your suggestion, I have added the missing ones

Reply:
When the type of an input is a list, I suggest adding the `Dict` type and specifying which elements are in that dict. This is better because it also describes the _structure_ of the dict. Here's a suggestion:

```python
def on_hpc_save(self, checkpoint: Dict[str, str]):
```

Reply:
@luiscape sounds good, I can improve it. For checkpoint I think it's not str. Should it be Dict[str, Any]?

Reply:
if you do not have restriction on the dictionary value, the `Any`